jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/commands/UserGroupCallbackTaskCommand.java

```diff
@@ -391,9 +411,10 @@ protected void doCallbackOperationForTaskDeadlines(Deadlines deadlines, TaskCont
                                         for(OrganizationalEntity recipient : recipients) {
                                             if(recipient instanceof User) {
                                                 doCallbackUserOperation(recipient.getId(), context);
-                                            }
-                                            if(recipient instanceof Group) {
+                                            } else if (recipient instanceof Group) {
                                                 doCallbackGroupOperation(recipient.getId(), context);
+                                            } else if (recipient instanceof Email) {
+                                                doCallbackEmailOperation(recipient.getId(), context);
                                             }
                                         }
                                     }
```

Comment:
This block (L407-420) is repeated also for End Deadlines. Perhaps, now it's a good opportunity to refactor it and extract it into a private method for enhancing maintainability.

Reply:
done

Reply:
ok